src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2450,9 +2451,9 @@ static bool updateContextAttributeItem
       ca->found = false;
     }
   }
-
-  /* Check aspects related with location */
-  if (!processLocationAtUpdateAttribute(currentLocAttrName, targetAttr, geoJson, &err, apiVersion, oe))
+  /* Check aspects related with location and date expiration */
+  if (!processLocationAtUpdateAttribute(currentLocAttrName, targetAttr, geoJson, &err, apiVersion, oe)
```

Comment:
Doing this way, a false in `!processDateExpirationAtUpdateAttribute(targetAttr, dateExpiration, replaceDate, &err, oe)` would cause reporting a bug related with location.

Thus, it would be better to add the checking for !processLocationAtUpdateAttribute(currentLocAttrName, targetAttr, geoJson, &err, apiVersion, oe) in a separated if clause.

Reply:
The code in the if clause, as far as i see, should be general in both cases. For this reason I kept a single if clause. In both cases it is printed the corresponding targetAttr name and error details.
On one hand, separating it in 2 clauses, ATM it would mean replicating the same code within both of them.
In the other hand, It would make sense to separate them if error handling will differ in the future for the two cases.

Reply:
Let me have a closer look in order to provide more precise feedback about this.

Reply:
Looking in detail, what is done if the `if ()` detects an error is to build this error:
```
     std::string details = std::string("action: UPDATE") +
                           " - entity: [" + eP->toString() + "]" +
                           " - offending attribute: " + targetAttr->getName() +
" - " + err;
```

and raise a bad input alarm:

```
alarmMgr.badInput(clientIp, err);
```

So, it seems to be a generic processing that could apply to both location and data expiration errors.

Maybe it would be a good idea to improve the `details` error to distinguish between both causes (thus, in that case, we would need probably two if clauses). However, let's leave it as it is now and check once the PR gets closer to finish (i.e. when the .test get ready) to decide what's better.

Reply:
NTC by the moment (I'll re-review on the new version of the code, rasing a comment if needed)